src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -788,9 +788,23 @@ static StructLayout captureStateLayout() {
          * <p>
          * Using this linker option when linking non-critical functions is likely to have adverse effects,
          * such as loss of performance, or JVM crashes.
+         * <p>
+         * Critical functions can optionally allow access to the Java heap. This allows a client to pass heap
+         * memory segments as addresses, where normally only off-heap memory segments would be allowed. The memory region
+         * inside the Java heap is exposed through a temporary native address that is valid for the duration of the
+         * function call. As such, these temporary addresses, or any addresses derived from them, should not be used
+         * after the function returns. Use of this mechanism is therefore only recommend when a function needs to do
+         * short-lived access to Java heap memory, and copying the relevant data to an off-heap memory segment would be
+         * prohibitive in terms of performance.
+         *
+         * @implNote As a consequence of allowing heap access, the JVM will either lock the garbage collector, or pin the
```

Comment:
This is a very low-level comment. Which is fine per se. But it doesn't say what does it mean for the developer using this functionality. I think you want to say that GC is impacted in some way or form. If so, please say.

Reply:
This comment can be removed, as we no longer lock the GC. The same restrictions apply as a 'regular' critical call that does not allow passing heap segments. The call should complete quickly.